### DISCUSSION TITLE
Laravel resource nested pagination

### DISCUSSION THREAD

**Discussion by mohammadhsn** [laravel/framework#46794] [General]
I have two eloquent models called `Product` and `Category.` They also have their own related HTTP resource classes.

In `CategoryController@show,` I have something like the below:

```php
class CategoryController
{
    public function show($id)
    {
        $category = Category::findOrFail($id);
        $category->setRelation('products', Product::where('category_id', $id)->paginate());
        return CategoryResource::make($category);
    }
}
```

My category resource shows the products as well:

```php

class CategoryResource
{
    public function toArray()
    {
        return [
            'id' => $this->resource->id,
            'products' => ProductResource::collection($this->whenLoaded('products'))
        ];
     }
}
```

I expect something like the below in the JSON response, but the pagination info doesn't exist inside products:
```json
{
    "id": 1,
    "products": [
        {
            "id": 1,
        },
        {
            "id": 2,
        }
    ]
}
```
What I expected from the JSON response:
```json
{
    "id": 1,
    "products": {
        "data": [
            {
                "id": 1,
            },
            {
                "id": 2,
            }
        ],
        "pagination": {
            // pagination stuff
        }
    }
}
```

**Comment by infureal** [comment]
Why you're not doing this instead?
```
GET api/categories
{
// collection of categories with pagination
}
```

then get an ID from category and go to:
```
GET api/categories/{id}/products
{
// collection of products with pagination
}
```

**Q: You need to response exactly all in once?**

**Comment by mohammadhsn** [comment]
Yes, I do

**Reply by Angel5a** [reply]
Then you can write your own resource collection, that will return pagination with your data. You can duplicate core code for that purpose. Or you can make a PR that will introduce ability to optionally include pagination into nested resources. See [docs](https://laravel.com/docs/10.x/eloquent-resources#wrapping-nested-resources) and source code.
